File: src/core/py/type.py
def core_type(a):
  if (type(a) is float):
    return 'num'
  if (type(a) is int):
    return 'num'
  elif (type(a) is bool):
    return 'bool'
  elif (type(a) is str):
    return 'string'
  elif (type(a) is list):
    return 'list'
  else:
    return 'unknown'

File: src/core/cpp/type.cpp
Box* core_type(Box *a) {
  if (0 == a->type) {
    return new Box("num");
  } else if (1 == a->type) {
    return new Box("string");
  } else if (2 == a->type) {
    return new Box("list");
  } else if (3 == a->type) {
    return new Box("list");
  } else {
    return new Box("unknown");
  }
}
